elasticapm/contrib/opentracing/tracer.py

```diff
@@ -0,0 +1,96 @@
+import warnings
+
+from opentracing import Format, InvalidCarrierException, SpanContextCorruptedException, UnsupportedFormatException
+from opentracing.scope_managers import ThreadLocalScopeManager
+from opentracing.tracer import ReferenceType
+from opentracing.tracer import Tracer as TracerBase
+
+import elasticapm
+from elasticapm import instrument, traces
+from elasticapm.conf import constants
+from elasticapm.contrib.opentracing.span import OTSpan, OTSpanContext
+from elasticapm.utils import compat, disttracing
+
+
+class Tracer(TracerBase):
+    _elasticapm_client_class = elasticapm.Client
+
+    def __init__(self, client_instance=None, config=None, scope_manager=None):
+        self._agent = client_instance or self._elasticapm_client_class(config=config)
+        if scope_manager and not isinstance(scope_manager, ThreadLocalScopeManager):
+            warnings.warn(
+                "Currently, the Elastic APM opentracing bridge only supports the ThreadLocalScopeManager. "
+                "Usage of other scope managers will lead to unpredictable results."
+            )
+        self._scope_manager = scope_manager or ThreadLocalScopeManager()
+        if self._agent.config.instrument:
+            instrument()
+
+    def start_active_span(
+        self,
+        operation_name,
+        child_of=None,
+        references=None,
+        tags=None,
+        start_time=None,
+        ignore_active_span=False,
+        finish_on_close=True,
+    ):
+        ot_span = self.start_span(
+            operation_name,
+            child_of=child_of,
+            references=references,
+            tags=tags,
+            start_time=start_time,
+            ignore_active_span=ignore_active_span,
+        )
+        scope = self._scope_manager.activate(ot_span, finish_on_close)
+        return scope
+
+    def start_span(
+        self, operation_name=None, child_of=None, references=None, tags=None, start_time=None, ignore_active_span=False
+    ):
+        if isinstance(child_of, OTSpanContext):
+            parent_context = child_of
+        elif isinstance(child_of, OTSpan):
+            parent_context = child_of.context
+        elif references and references[0].type == ReferenceType.CHILD_OF:
+            parent_context = references[0].referenced_context
+        else:
+            parent_context = None
+        transaction = traces.execution_context.get_transaction()
+        if not transaction:
+            trace_parent = parent_context.trace_parent if parent_context else None
+            transaction = self._agent.begin_transaction("opentracing", trace_parent=trace_parent)
```

Comment:
not sure how meaningful is this `opentracing` transaction type... the default `custom` seems just as good to me, but probably makes sense to align with other agents (maybe you did that already)